### ISSUE TITLE
[QUESTION] - Update a component

### ISSUE THREAD

**Issue by tramhao** [veeso/tui-realm#22]
I'm still learning the new API so probably my implementation have serious problem. Basically I save the playlist to a file, and load it when program start. I do it this way:
```
    pub fn run(&mut self) {
        self.model.init_terminal();
        assert!(self.model.load_playlist().is_ok());
 
```

But the problem is, the playlist is loaded and playable, but I cannot redraw the playlist component in this run function, because I don't have a view here. 

My question is: what's the proper way to do it? Send a msg? But how?

Sorry for my question but I spent two hours without progress. Thanks in advance.

**Comment by tramhao** [comment]
I have a sync_playlist function as below:
```
    pub fn sync_playlist(&mut self, view: &mut View<Id, Msg, NoUserEvent>) {
        let mut table: TableBuilder = TableBuilder::default();

        for (idx, record) in self.playlist_items.iter().enumerate() {
            if idx > 0 {
                table.add_row();
            }

            let duration = record.duration_formatted().to_string();
            let duration_string = format!("[{:^6.6}]", duration);

            let noname_string = "No Name".to_string();
            let name = record.name().unwrap_or(&noname_string);
            let artist = record.artist().unwrap_or(name);
            let title = record.title().unwrap_or("Unknown Title");

            table
                .add_col(TextSpan::new(duration_string.as_str()))
                .add_col(TextSpan::new(artist).fg(tuirealm::tui::style::Color::LightYellow))
                .add_col(TextSpan::new(title).bold())
                .add_col(TextSpan::new(record.album().unwrap_or("Unknown Album")));
        }
        if self.playlist_items.is_empty() {
            table.add_col(TextSpan::from("0"));
            table.add_col(TextSpan::from("empty playlist"));
            table.add_col(TextSpan::from(""));
            table.add_col(TextSpan::from(""));
        }

        let table = table.build();
        view.attr(
            &Id::Playlist,
            tuirealm::Attribute::Content,
            tuirealm::AttrValue::Table(table),
        )
        .ok();

        // if let Some(props) = self.view.get_props(COMPONENT_TABLE_PLAYLIST) {
        //     let props = TablePropsBuilder::from(props).with_table(table).build();
        //     let msg = self.view.update(COMPONENT_TABLE_PLAYLIST, props);
        //     self.update(&msg);
        // }
        // self.update_title_playlist();
    }

```

It require a view to render. But how can I get it in run of UI?

**Comment by tramhao** [comment]
I only found view in Update

**Comment by veeso** [comment]
I don't have the complete context of this case, so I'm not sure if I've got it right, but maybe one of these two systems could suit for you:

- Use a `redraw` flag in the Model and at each loop of the parent structure (e.g. Activity or Ui) check its value to redraw or not the ui.
- Use a `Task`. I'm using these in tuifeed and termscp and basically are Enum stored in a vec in the model which is consumed by the `Ui` or `Activity` after calling the `tick()` method on the `Application` in order to perform a high-level action caused by a `Msg`. For example I cannot fetch RSS sources in my Model, but I do it in the `Ui`. When I receive a `Msg` of kind `FetchSource` by a component, in the `update()` I set a task in the model `Task::FetchSource`, which is later consumed by the `Ui`.

Basically the View in the model should be used just for:

- Umounting
- Changing focus
- Remounting (Only if strictly necessary).

At the moment you cannot view() from the update (unless you create duplicated code). I'm thinking of creating a trait to do so, but I need to check whether it could actually work, due to lifetime issues.

**Comment by veeso** [comment]
I've thought about this issue and also considering my experience on tuifeed and on termscp, I've decided to return a `Vec` of `Msg` in the `tick()` method. So you can implement the `update()` function directly in the structure which holds the `Application` as you used to do in the older version.

I thought it could be good as a design and I still think it is, but it's really too uncomfortable to implement.

I'm going to release this tomorrow

**Comment by veeso** [comment]
1.1.0 Released.

Tick will now return the list of messages to process. Please, follow the example to see how to deal with it.

**Comment by tramhao** [comment]
Now it works and thanks!
